sympy/physics/control/lti.py

```diff
@@ -1300,6 +1302,50 @@ def to_expr(self):
             return Pow(self.den, -1, evaluate=False)
 
 
+class PIDController(Function):
+    """
+    A PID Controller class.
+
+    Parameters
+    ==========
+    KP : Expr
+        Proportional gain
+    KI : Expr
+        Integral gain
+    KD : Expr
+        Derivative gain
+    s : Symbol
+        The complex frequency variable
+
+    Examples
+    ========
+    >>> from sympy.abc import s
+    >>> from sympy.physics.control.lti import PIDController
+    >>> KP, KI, KD = 1, 0.1, 0.01
+    >>> pid = PIDController(KP, KI, KD, s)
+    >>> pid.transfer_function
+    TransferFunction(s**2*0.01 + s*1 + 0.1, s, s)
+    """
+
+    def __new__(cls, KP, KI, KD, s):
+        if not s.is_Symbol:
+            raise ValueError("s must be a symbol")
+        cls.KP = KP
+        cls.KI = KI
+        cls.KD = KD
+        cls.s = s
+
+        tf = TransferFunction(s*KD + KP + KI/s, 1, s)
+
+        return super().__new__(cls, tf)
+
+    @property
+    def transfer_function(self):
```

Comment:
You'll come across `.doit()` to convert other classes to `TransferFunction`. I would like that to be added instead of a function named `transfer_function` itself.

Reply:
sir so do we need to create a custom .doit()?

Reply:
> sir so do we need to create a custom .doit()?

I believe we do here.